File: 프로그래머스/lv2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.kt
import java.util.*

class Solution {
    fun solution(queue1: IntArray, queue2: IntArray): Int {
        var answer: Int = 0
        
        
        val q1:Queue<Long> = LinkedList()
        val q2:Queue<Long> = LinkedList()
        
        for(idx in queue1.indices) {
            q1.add(queue1[idx].toLong())
            q2.add(queue2[idx].toLong())
        }
       
        var s1: Long = queue1.sum().toLong()
        var s2: Long = queue2.sum().toLong()
        
        var total:Long = s1 + s2
        var temp = 0L
        
        
        
        if(total%2==1L) return -1
        
        
    
        while(true) {
            
            if(answer>queue1.size*3) {
                answer =-1
                break
            }
            
            if(s1==s2) {
                break
            }
            if(s1>s2){
                temp = q1.remove()
                q2.add(temp)
                s1-=temp
                s2+=temp
                
                answer++
            }
            else {
                temp = q2.remove()
                q1.add(temp)
                
                s2-=temp
                s1+= temp
                answer++
            
            }
        }
        
        return answer
    }
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0

    sum_q = sum(queue1) + sum(queue2)
    limit = len(queue1)*2 +1
    
    if sum_q % 2 == 1:
        return -1
    
    q1 = deque(queue1)
    q2 = deque(queue2)
    sum_q1 =sum(q1)
    sum_q2 = sum(q2)
    
    while sum_q1!=sum_q2:
        if answer>limit:
            return -1
            
        if sum_q1 > sum_q2:
            temp = q1.popleft()
            q2.append(temp)
            sum_q1-=temp
            sum_q2+=temp
        else:
            temp = q2.popleft()
            q1.append(temp)
            sum_q2-=temp
            sum_q1+=temp
        answer +=1
    
    return answer